autoload/neomake/makers/ft/c.vim

```diff
@@ -39,3 +45,26 @@ function! neomake#makers#ft#c#gcc()
             \ '%f:%l: %m',
         \ }
 endfunction
+
+function! neomake#makers#ft#c#clangtidy()
+    " Default arguments.
+    let l:args = []
+
+    " Add user-defined arguments if some are set.
+    " The -p option followed by the path to the build directory is expected.
+    " That directory should contain the compile command database
+    " (compile_commands.json).
+    if exists("g:neomake_c_clangtidy_args_conf")
+        let l:args += g:neomake_c_clangtidy_args_conf
```

Comment:
This isn't needed. Neomake already provides a way to modify maker arguments through a configuration variable:

```
" This already works for any maker, see :help g:neomake_<ft>_<name>_<property>
let g:neomake_c_clangtidy_args = []
```